go/plugins/ollama/ollama.go

```diff
@@ -344,3 +388,31 @@ func concatMessages(input *ai.GenerateRequest, roles []ai.Role) string {
 	}
 	return sb.String()
 }
+
+// concatMessages translates a list of messages into a prompt-style format
+func concatImages(input *ai.GenerateRequest, roles []ai.Role) ([]string, error) {
+	roleSet := make(map[ai.Role]bool)
+	for _, role := range roles {
+		roleSet[role] = true // Create a set for faster lookup
+	}
+
+	var images []string
+
+	for _, message := range input.Messages {
+		// Check if the message role is in the allowed set
+		if roleSet[message.Role] {
+			for _, part := range message.Content {
+				if !part.IsMedia() {
```

Comment:
I worry that if a request is part text and part images, it will fail here.
I think we want to just subselect the images?

Reply:
Good catch!